excellent/functions/builtin.go

```diff
@@ -1518,23 +1519,55 @@ func TimeFromParts(env envs.Environment, hour, minute, second int) types.XValue
 // Array Functions
 //------------------------------------------------------------------------------------------
 
+// Slice extracts a sub-sequence of items from `array`.
+//
+// The returned items are those from `start` up to but not-including `end`. Indexes start at zero and a negative
+// end value counts back from the end.
+//
+//	@(slice(array("a", "b", "c"), 0, 2)) -> [a, b]
+//	@(slice(array("a", "b", "c"), 1, 3)) -> [b, c]
+//	@(slice(array("a", "b", "c"), 0, -1)) -> [a, b]
+//	@(slice(array("a", "b", "c"), 1)) -> [b, c]
+//	@(slice(array("a", "b", "c"), 10)) -> []
+//
+// @function slice(array, start, [,end])
+func Slice(env envs.Environment, array *types.XArray, args ...types.XValue) types.XValue {
+	start, xerr := types.ToInteger(env, args[0])
+	if xerr != nil {
+		return xerr
+	}
+	if start < 0 {
+		return types.NewXErrorf("must start with a positive index")
+	}
+
+	end := array.Count()
+	if len(args) == 2 {
```

Comment:
what if we have more than 2 items in args, I think we allow this to work with the second item and ignore the others

so I guess we use `len(args) >= 2`

Reply:
The function is wrapped with `InitialArrayFunction` which takes care of enforcing the number of allowed args